include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -1246,6 +1246,173 @@ namespace Differentiation
       const std::vector<std::pair<ExpressionType, ValueType>> &symbol_values,
       const bool force_cyclic_dependency_resolution = false);
 
+    /**
+     * Perform a single substitution sweep of a set of symbols into the given
+     * symbolic expression.
+     * The symbols in the @p expression that correspond to the entry keys
+     * of the @p substitution_map are substituted with the map entry's associated
+     * value.
+     * This substitution function may be used to give a set of symbolic
+     * variables either a numeric interpretation or some symbolic definition.
+     *
+     * @note It is not required that all symbolic expressions be fully resolved
+     * when using this function. In other words, partial substitutions are
+     * valid.
+     *
+     * @note This function call is typically expensive, as by default it performs a
+     * dictionary substitution for the symbols in the symbolic expression.
+     * Should the numerical values of some symbolic expression be derired, then
```

Comment:
I believe you didn't want to talk about multiple values for a single expression, did you?

```suggestion
     * Should the numerical value of some symbolic expression be desired, then
```